sycl/include/sycl/ext/oneapi/sub_group_mask.hpp

```diff
@@ -89,6 +91,37 @@ struct sub_group_mask {
     BitsType RefBit;
   };
 
+#if SYCL_EXT_ONEAPI_SUB_GROUP_MASK >= 2
+  sub_group_mask() : sub_group_mask(0, GetMaxLocalRangeSize()){};
+
+  sub_group_mask(unsigned long long val)
+      : sub_group_mask(0, GetMaxLocalRangeSize()) {
+    constexpr size_t BytesToCopy =
+        sizeof(Bits) < sizeof(val) ? sizeof(Bits) : sizeof(val);
+    // TODO: memcpy is not guaranteed to work in kernels. Find alternative.
+    std::memcpy(&Bits, &val, BytesToCopy);
```

Comment:
Wouldn't an assignment from `val` to `Bits` be sufficient here?

Reply:
That's probably good idea [a22890d](https://github.com/intel/llvm/pull/11195/commits/a22890d770d91b6cf53387b1b8b43440b2142d43)